### PR TITLE
User: Add goto link for the login page (ILIAS 10)

### PR DESCRIPTION
**Pull request by lscharmer**
Add goto link (`/go/auth/login` & `goto.php/auth/login`) which redirects to the login page.

Feature Request: https://docu.ilias.de/goto_docu_wiki_wpage_8233_1357.html

### CODE REVIEW

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]
Hi @lscharmer ,

thx for this PR. In my opinion, we need to display the link in the ILIAS footer. Please have a look at `ilPasswordAssitanceGUI`, you can adapt the code.

Best regards,
Michael

File: components/ILIAS/User/classes/class.ilObjUserGUI.php [hunk 1]
```diff
@@ -1896,6 +1896,11 @@ public static function _goto(string $a_target): void
             $ilCtrl->redirectByClass(['ilStartUpGUI', 'ilPasswordAssistanceGUI'], 'showUsernameAssistanceForm');
         } elseif ('pwassist' == $a_target) {
             $ilCtrl->redirectByClass(['ilStartUpGUI', 'ilPasswordAssistanceGUI'], '');
+        } elseif ('login' === $a_target) {
+            $ilCtrl->redirectToURL('/login.php?' . http_build_query([
```
Comment (mjansenDatabay) [inline components/ILIAS/User/classes/class.ilObjUserGUI.php:1900]: Do we need a `cmd=foce_login` query parameter here? IMO there are a few occurrences in our code base, where this parameter is appended to ensure the display of the login page, even it the user/initialisation state is "undefined".

**Review by mjansenDatabay — APPROVED** [review]

**Review by mjansenDatabay — APPROVED** [review]

File: components/ILIAS/Authentication/classes/StaticUrlHandler.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    public function getNamespace(): string
+    {
+        return 'auth';
+    }
+
+    public function handle(Request $request, Context $context, Factory $response_factory): Response
+    {
+        $additional_params = join('/', $request->getAdditionalParameters() ?? []);
```
Comment (mjansenDatabay) [inline components/ILIAS/Authentication/classes/StaticUrlHandler.php:51]: Why do we need this?

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]

File: components/ILIAS/Authentication/classes/StaticUrlHandler.php (hunk 2 above)
Comment (lscharmer) [reply components/ILIAS/Authentication/classes/StaticUrlHandler.php:51]: Duplicated line is removed

**Review by mjansenDatabay — APPROVED** [review]

**Pull request merged**